misc/bstr.h

```diff
@@ -56,9 +56,11 @@ static inline struct bstr bstrdup(void *talloc_ctx, struct bstr str)
     return r;
 }
 
+#define bstr0_s(s) (struct bstr){(unsigned char *)(s), (s) ? strlen(s) : 0}
```

Comment:
This isn't portable at all. If you're just trying to initialize a string literal then use `sizeof - 1` which is guaranteed to be a constant expression.

```c
#define bstr0lit(s) { (unsigned char *)(s), sizeof(s) - 1 }
```

Reply:
I know, but does mpv support any other compiler than clang or gcc? I don't think so and likely other issues would arise if such compilation were tried.

Either way, I can change to sizeof, it is better probably.

Reply:
> I know, but does mpv support any other compiler than clang or gcc?

Even clang and gcc will not build this if builtin optimizations are disabled (e.g `-fno-builtin`).

Also, one more thing, surrounding the arg with empty string would be a nice improvement since it'd detect if the macro is being used on non string-literals.

```c
// should only be used on string literals
#define bstr0lit(s) { (unsigned char *)("" s ""), sizeof(s) - 1 }
```

Reply:
ACK, will fix later.

Reply:
> I added a commit https://github.com/mpv-player/mpv/commit/0a78ed3dca58e57cf85fb57e71cff22e17e9759c to fix this.

Trying to support `bstr0lit(NULL)` is basically zero value. If you want null string, just do `{0}`.

Also, I don't really see any reason to keep the current `bstr0_s` macro around. If someone is looking for optimization/inlining just enable LTO really.